cpp/examples/tutorial_examples/compute_example.cc

```diff
@@ -0,0 +1,99 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <arrow/api.h>
+#include <arrow/compute/api.h>
+
+#include <iostream>
+
+arrow::Status RunMain() {
+  // Create a couple 32-bit integer arrays.
+  arrow::Int32Builder int32builder;
+  int32_t some_nums_raw[5] = {34, 624, 2223, 5654, 4356};
+  ARROW_RETURN_NOT_OK(int32builder.AppendValues(some_nums_raw, 5));
+  std::shared_ptr<arrow::Array> some_nums;
+  ARROW_ASSIGN_OR_RAISE(some_nums, int32builder.Finish());
+
+  int32_t more_nums_raw[5] = {75342, 23, 64, 17, 736};
+  ARROW_RETURN_NOT_OK(int32builder.AppendValues(more_nums_raw, 5));
+  std::shared_ptr<arrow::Array> more_nums;
+  ARROW_ASSIGN_OR_RAISE(more_nums, int32builder.Finish());
+
+  // Make a table out of our pair of arrays.
+  std::shared_ptr<arrow::Field> field_a, field_b;
+  std::shared_ptr<arrow::Schema> schema;
+
+  field_a = arrow::field("A", arrow::int32());
+  field_b = arrow::field("B", arrow::int32());
+
+  schema = arrow::schema({field_a, field_b});
+
+  std::shared_ptr<arrow::Table> table;
```

Comment:
nit, but is there any value to having the table in this example?

Reply:
That when I used Arrow initially, it wasn't immediately clear to me how to use a Table, so this ensures people know how to do that very common task.

Reply:
Got it (I read this before seeing the doc, sorry for the duplicate question)